eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -106,47 +106,8 @@ if ($LatestsSorted[0] -ne $Version) {
 
 if ($ReplaceLatestEntryTitle)
 {
-    # Remove empty sections from content
-    $sanitizedContent = @()
-    $sectionContent = @()
-    $sectionContentCount = 0
-    $latesVersionContent = $ChangeLogEntries[$LatestVersion].ReleaseContent
-
-    foreach ($line in $latesVersionContent)
-    {
-        if ($line.StartsWith("### ") -or $sectionContentCount -gt 0)
-        {
-            if ($line.StartsWith("#") -and $sectionContentCount -gt 1)
-            {
-                $sanitizedContent += $sectionContent
-                $sectionContent = @()
-                $sectionContentCount = 0
-            }
-    
-            if ($line.StartsWith("#") -and $sectionContentCount -eq 1)
-            {
-                $sectionContent = @()
-                $sectionContentCount = 0
-            }
-
-            $sectionContent += $line
-            if (-not [System.String]::IsNullOrWhiteSpace($line))
-            {
-                $sectionContentCount++
-            }
-        }
-        elseif ($sectionContent.Count -eq 0)
-        {
-            $sanitizedContent += $line
-        }
-    }
-    
-    if ($sectionContentCount -gt 1)
-    {
-        $sanitizedContent += $sectionContent
-    }
-
-    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -Content $sanitizedContent
+    $sanitizedChangelogEntry =  Remove-EmptySections -ChangeLogEntry $ChangeLogEntries[$LatestVersion]
```

Comment:
I'm still not sure we want to always clean up these sections whenever someone passes in ReplaceLatestEntryTitle. Should we add an additional flag to indicate we want to finalize change log which would do this?

Reply:
We could do that. It might require refactoring calls to the Update-ChangeLog.ps1 script especially for python and js.
As I understand it its fine to do this here, because replacing the latest entry is always a sign that one is preparing for a release, Am not aware of any situttion where that is not the case

Reply:
Can you please test Java? I want to make sure what we don't end up accidently removing all these sections for changelogs that aren't actually planning to ship.

Reply:
For example https://github.com/Azure/azure-sdk-for-java/blob/main/eng/versioning/update_versions.py#L173 when we update a version dependency in a file we might end up causing a lot of changelog files to be cleaned.

Reply:
That Java line actually show how the workflow from the perspective of the application. Its either a version increment which adds a new changelog entry, or it is updating the latest entry which implies one is preparing for a release. Any other changelog changes will happen manually without the use of the tooling. The parameter name was formerly [ReplaceLatestEntry](https://github.com/Azure/azure-sdk-tools/blob/e6d24dce8fe81b327ba806cf6f8ef2d1935f3eca/eng/common/scripts/Update-ChangeLog.ps1#L16) I actually changed it to `ReplaceLatestEntryTitle` because that was all it was doing.


The only other case of update coming through this script is if `ReplaceLatestEntryTitle` is no passed but the version is present in the changelog. This branch will update the `$ReleaseStatus` for any entry in the changelog. We almost never make use of this though.